pacman/utilities/file_format_converters/convert_to_java_machine.py

```diff
@@ -0,0 +1,137 @@
+from collections import OrderedDict
+import json
+
+from pacman.utilities import file_format_schemas
+from spinn_utilities.progress_bar import ProgressBar
+from spinn_machine.router import Router
+
+
+class ConvertToJavaMachine(object):
+    """ Converter from memory machine to java machine
+    """
+
+    __slots__ = []
+
+    def __call__(self, machine, file_path):
+        """
+        Runs the code to write the machine in Java readable json.
+
+        :param machine: Machine to convert
+        :param file_path: Location to write file to. Warning will overwrite!
+        """
+        progress = ProgressBar(
+            (machine.max_chip_x + 1) * (machine.max_chip_y + 1) + 2,
+            "Converting to JSON machine")
+
+        for chip in machine.chips:
+            if (chip.ip_address is None):
+                s_monitors = chip.n_processors - chip.n_user_processors
+                s_router_entries = chip.router.n_available_multicast_entries
+                s_router_clock_speed = chip.router.clock_speed
+                s_sdram = chip.sdram.size
+                s_virtual = chip.virtual
+                s_tags = chip._tag_ids
+                break
+
+        chip = machine.boot_chip
```

Comment:
these 6 lines can be removed and put in directly, as with below change they become redundant

Reply:
NO they can not.
This find one none ethernet chips gets its values and then kills the for loop

The lines below use the ethernet chip and save to different fields.